cajparser.py

```diff
@@ -493,6 +540,12 @@ def _parse_hn(self):
             print("Text:\n", page_data.texts)
             print("Figures:\n", page_data.figures)
             current_offset = page_data_offset + size_of_text_section
+
+            if images_per_page > len(page_data.figures) or (squared and images_per_page > 1):
+                from math import sqrt
```

Comment:
Again, no need to see if it is square. You just want the smaller of two if they don't agree.